cognite_toolkit/_cdf_tk/commands/dump_resource.py

```diff
@@ -90,54 +90,59 @@ def __init__(self, client: ToolkitClient, identifier: DataModelId | None = None)
         self.space_ids: set[str] = set()
 
     def _interactive_select(self) -> DataModelId:
-        include_global = False
-        spaces = self.client.data_modeling.spaces.list(limit=-1, include_global=include_global)
-        selected_space: str = questionary.select(
-            "In which space is your data model located?", [space.space for space in spaces]
-        ).ask()
-
         data_model_ids = self.client.data_modeling.data_models.list(
-            space=selected_space, all_versions=False, limit=-1, include_global=include_global
+            all_versions=False, limit=-1, include_global=False
         ).as_ids()
-
-        if not data_model_ids:
-            raise ToolkitMissingResourceError(f"No data models found in space {selected_space}")
+        available_spaces = sorted({model.space for model in data_model_ids})
+        if not available_spaces:
+            raise ToolkitMissingResourceError("No data models found")
+        if len(available_spaces) == 1:
+            selected_space = available_spaces[0]
+        else:
+            selected_space = questionary.select("In which space is your data model located?", available_spaces).ask()
+        data_model_ids = sorted(
+            [model for model in data_model_ids if model.space == selected_space], key=lambda model: model.as_tuple()
+        )
 
         selected_data_model: DataModelId = questionary.select(
-            "Which data model would you like to dump?", [Choice(f"{model!r}", value=model) for model in data_model_ids]
+            "Which data model would you like to dump?",
+            [
+                Choice(f"{model_id!r}", value=model_id)
+                for model_id in sorted(data_model_ids, key=lambda model: model.as_tuple())
+            ],
         ).ask()
 
-        data_models = self.client.data_modeling.data_models.list(
-            space=selected_space,
-            all_versions=True,
-            limit=-1,
-            include_global=include_global,
-            inline_views=False,
+        retrieved_models = self.client.data_modeling.data_models.retrieve(
+            (selected_data_model.space, selected_data_model.external_id), inline_views=False
         )
-        data_model_ids = data_models.as_ids()
-        data_model_versions = [
-            model.version
-            for model in data_model_ids
+        if not retrieved_models:
+            # This happens if the data model is removed after the list call above.
+            raise ToolkitMissingResourceError(f"Data model {selected_data_model} not found")
+        if len(retrieved_models) == 1:
+            self.data_model = retrieved_models[0]
+            return selected_data_model
+        models_by_version = {
+            model.version: model
+            for model in retrieved_models
             if (model.space, model.external_id) == (selected_data_model.space, selected_data_model.external_id)
             and model.version is not None
```

Comment:
Is the condition on (space, external_id) needed, given you pass space and external_id into the `retrieve` call?

Reply:
You are correct.